File: components/mutations/SetFeedOptions.ts
import { Environment,graphql } from "react-relay"

import { commitMutationAsync } from "./commitMutationAsync"

const mutation = graphql`
  mutation SetFeedOptionsMutation($input: SetFeedOptionsInput!) {
    setFeedOptions(input: $input) {
      feed {
        id
        autopost
      }
    }
  }
`

export interface SetFeedOptionsInput {
  id: string
  autopost: boolean
}

export async function setFeedOptions(
  environment: Environment,
  input: SetFeedOptionsInput
): Promise<void> {
  const variables = { input }
  await commitMutationAsync(environment, {
    mutation,
    variables,
  })
}

File: ios/Courier/Courier/Mutations/SetFeedOptions.swift
import RelaySwiftUI
import CourierGenerated

private let mutation = graphql("""
mutation SetFeedOptionsMutation($input: SetFeedOptionsInput!) {
  setFeedOptions(input: $input) {
    feed {
      id
      autopost
    }
  }
}
""")

extension Mutation.Mutator where Operation == SetFeedOptionsMutation {
    func commit(id: String, autopost: Bool) {
        commit(
            variables: .init(input: .init(id: id, autopost: autopost)),
            optimisticResponse: [
                "setFeedOptions": [
                    "feed": [
                        "id": id,
                        "autopost": autopost,
                    ],
                ],
            ]
        )
    }
}
